### ISSUE TITLE
OCI: fail to download public bundles via without credentials

### ISSUE THREAD

**Issue by carabasdaniel** [open-policy-agent/opa#5902]
With the latest changes in v0.52.0 the OCI downloader returns a 401 error from ghcr.io when trying to download a public policy image.

Example configuration (example-file.yaml) :
```
services:
  ghcr-registry:
    url: https://ghcr.io/
    type: oci

bundles:
  acme:
    service: ghcr-registry
    resource: ghcr.io/carabasdaniel/testpol:latest
```

## Steps To Reproduce
`opa run -c example-file.yaml `

Logs:
```
{"level":"error","msg":"Bundle load failed: failed to pull ghcr.io/carabasdaniel/testpol:latest: download for 'ghcr.io/carabasdaniel/testpol:latest' failed: failed to resolve ghcr.io/carabasdaniel/testpol:latest: pulling from host ghcr.io failed with status code [manifests latest]: 401 Unauthorized","name":"acme","plugin":"bundle","time":"2023-05-04T20:04:22+03:00"}
```

## Expected behavior
Load bundle successfully from public image.

## Additional context
I suspect the problem might be related to the authorizer used here https://github.com/open-policy-agent/opa/blob/main/download/oci_download.go#L333 when there are no credentials/plugins configured.

**Comment by ashutosh-narkar** [comment]
@DerGut can you please look into this?

**Comment by DerGut** [comment]
I created a PR to fix it in https://github.com/open-policy-agent/opa/pull/5906.

Interestingly, we had a conversation about this exact code path in https://github.com/open-policy-agent/opa/pull/5560#discussion_r1160980752 which I removed with https://github.com/open-policy-agent/opa/commit/c5428d9f4df2032d879c2f0156555b7ec622c6ba.
Seems like docker still uses the auth challenges for public repos. I expected it to use no authorization instead but I couldn't find any docs on it.

The PR adds that logic back in.
